include/api.php

```diff
@@ -681,59 +682,112 @@ function api_item_get_user(&$a, $item) {
 		return (array($status_user, $owner_user));
 	}
 
-
 	/**
-	 * @brief transform $data array in xml without a template
+	 * @brief walks recursively through an array with the possibility to change value and key
+	 *
+	 * @param array $array The array to walk through
+	 * @param string $callback The callback function
 	 *
-	 * @param array $data
-	 * @return string xml string
+	 * @return array the transformed array
 	 */
-	function api_array_to_xml($data, $ename="") {
-		$attrs="";
-		$childs="";
-		if (count($data)==1 && !is_array($data[array_keys($data)[0]])) {
-			$ename = array_keys($data)[0];
-			$ename = trim($ename,'$');
-			$v = $data[$ename];
-			return "<$ename>$v</$ename>";
-		}
-		foreach($data as $k=>$v) {
-			$k=trim($k,'$');
-			if (!is_array($v)) {
-				$attrs .= sprintf('%s="%s" ', $k, $v);
+	function api_walk_recursive(array &$array, callable $callback) {
+
+		$new_array = array();
+
+		foreach ($array as $k => $v) {
+			if (is_array($v)) {
+				if ($callback($v, $k))
+					$new_array[$k] = api_walk_recursive($v, $callback);
 			} else {
-				if (is_numeric($k)) $k=trim($ename,'s');
-				$childs.=api_array_to_xml($v, $k);
+				if ($callback($v, $k))
+					$new_array[$k] = $v;
 			}
 		}
-		$res = $childs;
-		if ($ename!="") $res = "<$ename $attrs>$res</$ename>";
-		return $res;
+		$array = $new_array;
+
+		return $array;
+	}
+
+	/**
+	 * @brief Callback function to transform the array in an array that can be transformed in a XML file
+	 *
+	 * @param variant $item Array item value
+	 * @param string $key Array key
+	 *
+	 * @return boolean Should the array item be deleted?
+	 */
+	function api_reformat_xml(&$item, &$key) {
+		if (is_bool($item))
+			$item = ($item ? "true" : "false");
+
+		if (substr($key, 0, 10) == "statusnet_")
+			$key = "statusnet:".substr($key, 10);
+		elseif (substr($key, 0, 10) == "friendica_")
+			$key = "friendica:".substr($key, 10);
+		//else
+		//	$key = "default:".$key;
+
+		return true;
+	}
+
+	/**
+	 * @brief Creates the XML from a JSON style array
+	 *
+	 * @param array $data JSON style array
+	 * @param string $root_element Name of the root element
+	 *
+	 * @return string The XML data
+	 */
+	function api_create_xml($data, $root_element) {
+		$childname = key($data);
+		$data2 = array_pop($data);
+		$key = key($data2);
+
+		$namespaces = array("" => "http://api.twitter.com",
+					"statusnet" => "http://status.net/schema/api/1/",
+					"friendica" => "http://friendi.ca/schema/api/1/",
+					"georss" => "http://www.georss.org/georss");
+
+		/// @todo Auto detection of needed namespaces
```

Comment:
I think it's not needed. Just always put in namespace declarations and make the parser happy.. The parser don't complain on unused declared ns, but will stop on used undeclared ones...